examples/Migration/CampaignManagement/CreateCompleteCampaignGoogleAdsApiOnly.php

```diff
@@ -0,0 +1,499 @@
+        );
+
+        printf("Added budget named '%s'%s", $newCampaignBudget->getName()->getValue(), PHP_EOL);
+
+        return $newCampaignBudget;
+    }
+
+    /**
+     * Gets a campaign budget.
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param string $customerId the client customer ID without hyphens
+     * @param string $resourceName the resource name of the campaign budget to retrieve
+     * @return CampaignBudget the campaign budget
+     */
+    private static function getCampaignBudget(
+        GoogleAdsClient $googleAdsClient,
+        string $customerId,
+        string $resourceName
+    ) {
+        $googleAdsServiceClient = $googleAdsClient->getGoogleAdsServiceClient();
+        $query = "SELECT campaign_budget.id, campaign_budget.name, " .
+                 "campaign_budget.resource_name from campaign_budget where " .
+                 "campaign_budget.resource_name='$resourceName'";
+
+        $response =
+            $googleAdsServiceClient->search($customerId, $query, ['pageSize' => self::PAGE_SIZE]);
+
+        return $response->getIterator()->current()->getCampaignBudget();
+    }
+
+    /**
+     * Creates a campaign.
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param string $customerId the client customer ID without hyphens
+     * @param CampaignBudget $campaignBudget the campaign budget
+     * @return Campaign the newly created campaign
+     */
+    private static function createCampaign(
+        GoogleAdsClient $googleAdsClient,
+        string $customerId,
+        CampaignBudget $campaignBudget
+    ) {
+        $trueValue = new BoolValue(['value' => true]);
+        $falseValue = new BoolValue(['value' => false]);
+
+        $startDate = new StringValue(['value' => date('Ymd', strtotime('+1 day'))]);
+        $endDate = new StringValue(['value' => date('Ymd', strtotime('+1 month'))]);
+
+        // Configures the campaign network options.
+        $networkSettings = new NetworkSettings([
+            'target_google_search' => $trueValue,
+            'target_search_network' => $trueValue,
+            'target_content_network' => $falseValue,
+            'target_partner_search_network' => $falseValue
+        ]);
+
+        $budgetResourceName = $campaignBudget->getResourceName();
+
+        $campaign = new Campaign([
+            'name' => new StringValue(['value' => 'Interplanetary Cruise #' . uniqid()]),
+            'advertising_channel_type' => AdvertisingChannelType::SEARCH,
+            // Recommendation: Set the campaign to PAUSED when creating it to prevent
+            // the ads from immediately serving. Set to ENABLED once you've added
+            // targeting and the ads are ready to serve.
+            'status' => CampaignStatus::PAUSED,
+            // Sets the bidding strategy and budget.
+            'manual_cpc' => new ManualCpc(),
+            'campaign_budget' => $budgetResourceName,
```

Comment:
Please inline this. No need to define $budgetResourceName.

Reply:
Done.